pkg/reconciler/autoscaling/hpa/resources/keda.go

```diff
@@ -138,5 +142,46 @@ func DesiredScaledObject(pa *autoscalingv1alpha1.PodAutoscaler, config *autoscal
 		}
 	}
 
-	return &sO
+	return &sO, nil
+}
+
+func getDefaultPrometheusTrigger(annos map[string]string, address string, query string, threshold string, ns string) (*v1alpha1.ScaleTriggers, error) {
+	trigger := v1alpha1.ScaleTriggers{
+		Type: "prometheus",
+		Metadata: map[string]string{
+			"serverAddress": address,
+			"query":         query,
+			"threshold":     threshold,
+			// This is mandatory for OCP
```

Comment:
I'm not sure what this comment tells me? Is it not mandatory for upstream? If so should we handle it differently somehow?

Reply:
This is optional upstream: https://keda.sh/docs/2.14/scalers/prometheus

> namespace - A namespace that should be used for namespaced queries. These are required by some highly available Prometheus setups, such as [Thanos](https://thanos.io/). (Optional)

Downstream though for the OCP monitoring thingy:

https://docs.openshift.com/container-platform/4.16/nodes/cma/nodes-cma-autoscaling-custom-trigger.html

> 	Optional: Specifies the namespace of the object you want to scale. This parameter is mandatory if using OpenShift Container Platform monitoring as a source for the metrics.